os/arch/arm/src/amebad/amebad_watchdog_lowerhalf.c

```diff
@@ -0,0 +1,482 @@
+static int amebad_wdg_start(FAR struct watchdog_lowerhalf_s *lower);
+static int amebad_wdg_stop(FAR struct watchdog_lowerhalf_s *lower);
+static int amebad_wdg_keepalive(FAR struct watchdog_lowerhalf_s *lower);
+static int amebad_wdg_getstatus(FAR struct watchdog_lowerhalf_s *lower, FAR struct watchdog_status_s *status);
+static int amebad_wdg_settimeout(FAR struct watchdog_lowerhalf_s *lower, uint32_t timeout);
+static xcpt_t amebad_wdg_capture(FAR struct watchdog_lowerhalf_s *lower, xcpt_t handler);
+static int amebad_wdg_ioctl(FAR struct watchdog_lowerhalf_s *lower, int cmd, unsigned long arg);
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+const struct watchdog_ops_s g_wdgops = {
+	.start = amebad_wdg_start,
+	.stop = amebad_wdg_stop,
+	.keepalive = amebad_wdg_keepalive,
+	.getstatus = amebad_wdg_getstatus,
+	.settimeout = amebad_wdg_settimeout,
+	.capture = amebad_wdg_capture,
+	.ioctl = amebad_wdg_ioctl,
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+/****************************************************************************
+ * Name: amebad_wdg_start
+ *
+ * Description:
+ *   Start the watchdog timer, resetting the time to the current timeout,
+ *
+ * Input Parameters:
+ *   lower - A pointer the publicly visible representation of the "lower-half"
+ *           driver state structure.
+ *
+ * Returned Values:
+ *   Zero on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+static int amebad_wdg_start(FAR struct watchdog_lowerhalf_s *lower)
+{
+	struct amebad_wdg_lowerhalf_s *priv = (struct amebad_wdg_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
+	if (!priv) {
+		return -ENODEV;
+	}
+	if (priv->wdt_status == WDT_STOP) {
+		/* Start watchdog timer */
+		watchdog_start();
+		priv->wdt_status = WDT_START;
+		priv->start_tick = SYSTIMER_TickGet();	// Record time right after wdt start
+		priv->on_pause_duration = 0;	// Reset on_pause_duration
+	}
+
+	return OK;
+}
+
+/****************************************************************************
+ * Name: amebad_wdg_stop
+ *
+ * Description:
+ *   Stop the watchdog timer
+ *
+ * Input Parameters:
+ *   lower - A pointer the publicly visible representation of the "lower-half"
+ *           driver state structure.
+ *
+ * Returned Values:
+ *   Zero on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+static int amebad_wdg_stop(FAR struct watchdog_lowerhalf_s *lower)
+{
+	struct amebad_wdg_lowerhalf_s *priv = (struct amebad_wdg_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
+	if (!priv) {
+		return -ENODEV;
+	}
+
+	/* Stop watchdog timer */
+	watchdog_stop();
```

Comment:
could you check started?
```
if (isStarted) {
   watchdog_stop()
   return OK;
}

return other errocode;

Reply:
My idea about this stop function is that it can be used at any moment.
Your suggested changes will ignore the wdt status of **ON_PAUSE**.

Although, it can be changed to `if(priv->wdt_status != WDT_STOP)` if it is necessary to feedback errorcode when multiple stop functions are called from upper layer.